types/react/test/hooks.tsx

```diff
@@ -263,12 +266,27 @@ function useEveryHook(ref: React.Ref<{ id: number }> | undefined): () => boolean
     // make sure the generic argument does reject actual potentially undefined inputs
     // @ts-expect-error
     React.useState<number>(undefined)[0];
+    // don't allow void calls to update the state with the default overload
+    // @ts-expect-error
```

Comment:
Doesn't this contradict the PR title

> allow useState setter void calls when undefined

?

Reply:
The default overload is the one with the initial state. This call shouldn't be allowed when the user is using an initial state.

Reply:
Let's update the comment then. The default overload may change and there's possibly no notion of a "default overload" anyway.

Actually, the comment isn't even useful then. It just restates what's happening not why.

Reply:
I reworded the comment to explain better when this shouldn't be allowed. I hope that's better. Thanks.